homeassistant/components/hive/switch.py

```diff
@@ -25,50 +35,37 @@ async def async_setup_entry(
     devices = hive.session.deviceList.get("switch")
     entities = []
     if devices:
-        for dev in devices:
-            entities.append(HiveDevicePlug(hive, dev))
+        for description in SWITCH_TYPES:
+            for dev in devices:
+                if dev["hiveType"] == description.key:
+                    entities.append(HiveSwitch(hive, dev, description))
     async_add_entities(entities, True)
 
 
-class HiveDevicePlug(HiveEntity, SwitchEntity):
+class HiveSwitch(HiveEntity, SwitchEntity):
     """Hive Active Plug."""
 
-    @property
-    def unique_id(self):
-        """Return unique ID of entity."""
-        return self._unique_id
-
-    @property
-    def device_info(self) -> DeviceInfo | None:
-        """Return device information."""
-        if self.device["hiveType"] == "activeplug":
-            return DeviceInfo(
-                identifiers={(DOMAIN, self.device["device_id"])},
-                manufacturer=self.device["deviceData"]["manufacturer"],
-                model=self.device["deviceData"]["model"],
-                name=self.device["device_name"],
-                sw_version=self.device["deviceData"]["version"],
-                via_device=(DOMAIN, self.device["parentDevice"]),
-            )
-        return None
-
-    @property
-    def name(self):
-        """Return the name of this Switch device if any."""
-        return self.device["haName"]
+    entity_description: SwitchEntityDescription
+
+    def __init__(self, hive, hive_device, description):
+        """Intiate Hive Switch."""
+        super().__init__(hive, hive_device)
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self.device["device_id"])},
+            manufacturer=self.device["deviceData"]["manufacturer"],
+            model=self.device["deviceData"]["model"],
+            name=self.device["device_name"],
+            sw_version=self.device["deviceData"]["version"],
+            via_device=(DOMAIN, self.device["parentDevice"]),
+        )
+        self._attr_name = self.device["haName"]
+        self.entity_description = description
 
     @property
     def available(self):
         """Return if the device is available."""
         return self.device["deviceData"].get("online")
 
-    @property
-    def extra_state_attributes(self):
-        """Show Device Attributes."""
-        return {
-            ATTR_MODE: self.attributes.get(ATTR_MODE),
```

Comment:
Removing this state attribute is a breaking change. We need to mention this in the breaking change paragraph in the PR description.